test/connectivity/README.md

```diff
@@ -69,3 +72,106 @@ Network offline/online should not affect this test result since Edge hub stores
+
+##### Test Mechanism: 
+	1. The Direct Method Cloud Sender invokes a test direct method to the IoT Hub
+	2. The IoT Hub forwards the direct method to Edge Agent
+	3. The Edge Agent has a pre-implemented "ping" method with a response.
+	4. The Direct Method Cloud Sender receives a response from IoT Hub
+	5. The Direct Method Cloud Sender notifies Test Analyzer regarding its direct method response
+Test Analyzer compile a report from Direct Method Cloud Sender, Direct Method Receiver, Network Controller and determine if the test case is passed or failed. 
+
+#### Cloud to Device Message Test
+
+![Connectivity Test Diagram](./images/ConnectivityTest_C2D_Message.png)
+
+The goal of the test scheme is to verify the recoverability of the cloud to device messaging once the device goes offline. 
+##### Test Specific Components:
+	1. Message Sender is a Azure service client that is deployed into a test agent as a container module. The responsibility of the module is to use the Azure API to trigger IoT Hub to send a message to device along with reporting to Test Result Coordinator once a message is sent. 
+	2. Message Receiver is a device client that is wrapped in a module container for a deployment into a test agent. The device module behaves as a target receiving end of the message sending from Message Sender and is responsible for reporting to the Test Result Coordinator once a message is received.
+
+##### Test Implementation Assumption:
+	1. The cloud takes less than 15 minutes to forward all of messages from the cloud to device when the Edge Hub is online.
+	2. Each cycle of Network Controller online-offline period must be longer than the Edge Hub recovering period
+	3. Since the Message Sender uses service client to send a message to the cloud from the host network, it will always report success even if the message has yet gotten to the Message Receiver. If the Message Sender fails to send a message, the message sequencing ID should NOT be increased; instead, a failed message with the same sequence number should be resent.
+	
+##### Test Mechanism: 
+	1. The Message Sender invokes the IoT Hub to send  a message to a targeted device
+	2. The Message Sender notify the Test Result Coordinator that a message has been sent.
+	3. The IoT Hub sends a message to Edge Hub addressing a targeted device
+	4. The Edge Hub forward the message to a Message Receiver device module.
+	5. The Message Receiver notifies that the message successfully received to the Test Result Coordinator
+	6. The Message Receiver replies that the message successfully received to the Edge Hub
+	7. The Edge Hub forward the Message Receiver's response back to IoT Hub
+	8. The IoT Hub forward the response to Message Sender.
+
+##### Passing Criteria: 
+All the messages sent by Message Sender is received in order at Message Receiver within (EdgeHub_Recover_Time) + 15 minutes.
```

Comment:
what is edgehub_recover_time?

Reply:
It is how long edgeHub realize it is online/offline. This number can change depending on the SDK; hence, I'm not going to hard-coded it here.